### PR TITLE
Add custom constructors

### PR DESCRIPTION
**Pull request by hausdorff**

### CODE REVIEW

**Comment by AntonioMeireles** [comment]
Hi all! (just curious) 
what's the ETA / status of this specific one ? simplifies / smoothens a few things ... (and thanks / keep the great work!) 

TIA

**Comment by hausdorff** [comment]
@AntonioMeireles Hey, sorry, just seeing this now. We should probably try to land it soon, I think, sometime this week.

File: ksonnet-gen/kubeversion/data.go [hunk 1]
```diff
@@ -72,13 +72,82 @@ var versions = map[string]versionData{
...
+			"io.k8s.kubernetes.pkg.apis.extensions.v1beta1.StatefulSet":        v1beta1StatefulSet,
+			"io.k8s.kubernetes.pkg.apis.extensions.v1beta1.StatefulSetList":    objectList,
+
+			//
+			// Authentication namespace.
+			//
+
+			"io.k8s.kubernetes.pkg.apis.authentication.v1.TokenReview": []CustomConstructorSpec{
+				newConstructor(
+					"new",
```
Comment (jessicayuen) [inline ksonnet-gen/kubeversion/data.go]: There seems to be a mandatory "new" as the first param for every call to newConstructor -- what is this for?

**Comment by hausdorff** [comment]
@AntonioMeireles we've merged this. BTW, have a look at the improvements coming down the pike for beta.3:

https://github.com/ksonnet/ksonnet-lib/pull/78#issue-259903981

**Comment by AntonioMeireles** [comment]
@hausdorff thanks! watching this space closely ;-)

**Pull request merged**